lib/util/usedPropTypes.js

```diff
@@ -429,6 +497,12 @@ module.exports = function usedPropTypesInstructions(context, components, utils)
 
     FunctionExpression: handleFunctionLikeExpressions,
 
+    'FunctionDeclaration:exit': propVariables.popScope,
+
+    'ArrowFunctionExpression:exit': propVariables.popScope,
+
+    'FunctionExpression:exit': propVariables.popScope,
```

Comment:
do these not need to retain their receiver? (ie, `(...args) => propVariables.popScope(...args)`)

Reply:
No because `createPropVariables` uses closure pattern.

Reply:
In that case, could we immediately destructure out propVariables rather than using it as a namespace?